Cargo.toml

```diff
@@ -22,8 +22,7 @@ bytes = "1"
 argh = "0.1"
 crossterm = "0.27"
 flate2 = "1.0"
-futures = "0.3.29"
-futures-core = "0.3"
+futures-lite = "2.3.0"
 futures-util = { version = "0.3", default-features = false }
```

Comment:
We probably do want to get rid of futures-util. You'll notice that we have places like `src/service/layer/trace_err.rs` that make use of it for stuff like `::map`, but honestly you can just get rid of that, and instead make it a regular async method impl (using the `async` syntax and await the future so yoiu can just deal with the result normally.

That's just one example. But just so you know, you can use what's in `futures_lite/futures_core` if there's a replacement, but otherwise it might just mean you need to just do it differently, as shown in this example